cmd/clusterctl/hack/local-overrides.py

```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/env python
+
+# Copyright 2015 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+###################
+
+# local-overrides.py takes in input a list of provider and, for each of them, generates the components YAML from the
+# local repositories (the GitHub repositories clone), and finally stores it in the clusterctl local override folder
+
+# prerequisites:
+
+# - the script should be executed from sigs.k8s.io/cluster-api/ by calling cmd/clusterctl/hack/local-overrides.py
+# - there should be a sigs.k8s.io/cluster-api/clusterctl-settings.json file with the list of provider for which
+#   the local overrides should be generated and the list of provider repositories to be included (on top of cluster-api).
+# {
+#    "providers": [ "cluster-api", "kubeadm", "aws"],
+#    "provider_repos": ["../cluster-api-provider-aws"]
+# }
+# - for each additional provider repository there should be a sigs.k8s.io/<provider_repo>/clusterctl-settings.json file e.g.
+# {
+#   "name": "aws",
+#   "config": {
+#     "componentsFile": "infrastructure-components.yaml",
+#     "nextVersion": "v0.5.0",
+#     "type": "InfrastructureProvider"
+# }
+
+###################
+
+import json
+import subprocess
+import os
+import errno
+
+settings = {}
+
+providers = {
+      'cluster-api': {
+              'componentsFile': 'core-components.yaml',
+              'nextVersion': 'v0.3.0',
+              'type': 'CoreProvider',
+      },
+      'kubeadm': {
```

Comment:
Should this be called `kubeadm-bootstrap` (would match what's in the Tiltfile)?

Reply:
aligned